woocommerce/woocommerce-android
# Issue [5973] Calculate percentage fee

Closes: #5973

### Description
Please don't merge this PR until it is approved by design

This PR aims to fix the loss of the percentage state by adding some clarification to what the app is doing (calculate the fee as a percentage but save as an amount). Now when the user switches to calculating the fee as a percentage, the user can also see the fee amount result. Also, it removes the fee tax total from the fee percentage calculation for a more accurate result.

### Testing instructions
#### Test 1
1. Orders -> New Order (+) -> Create Order
2. Add fee -> Check that **calculate percentage** is hided
----
#### Test 2
1. Orders -> New Order (+) -> Create Order
2. Add product -> Select product
4. Add fee -> Check calculate percentage is shown
5. Press **calculate percentage** -> Enter a percentage -> Check the calculated amount is shown and updates accordingly
6. Switch calculate percentage several times
----
#### Test 3
1. Orders -> New Order (+) -> Create Order
2. Add product -> Select product
3. Add fee -> Check calculate percentage is shown
4. Press calculate percentage -> Enter a percentage -> Press Done
5. Remove the product (-) -> Press remove order from product
6. Add fee -> Check that **calculate percentage** is hided
----
#### Test 4
Run OrderCreationViewModelTest
Run OrderCreationFeeViewModelTest

### Images/gif
#### Final result

https://user-images.githubusercontent.com/18119390/165353286-1fd4e521-45a5-4052-8320-675edd290883.mp4


#### First iteration
https://user-images.githubusercontent.com/18119390/161331467-263b0c26-b299-45e9-bf01-36a0971a210e.mp4

#### Evidence
When calculating user fees, fees' taxes should be removed from the order's total.

In this example, I started by adding a product that cost 25.00, and with taxes, the order total became 25,50
Applying a fee of 10% turns into a fee of the 10% of 25,50 -> 2,55
Now, if I tried to modify that fee to the same amount (remove the 10% and set 10% again). As the percentage calculation includes the fee tax the new result is 2,74 and not the initial 2,55

https://user-images.githubusercontent.com/18119390/165530939-4108472f-5a5a-4aa0-a135-fa72116f40d0.mp4
